src/Application.spec.js

```diff
@@ -59,6 +59,35 @@ describe('Application', () => {
       });
     });
 
+    it('should call preload in the correct order for plugins with dependencies', () => {
+      let aPreloadCalled = false;
+      const a = new ApplicationPlugin({ name: 'a' });
+      // a preload that takes some time
+      a.preload = () => {
+        return new Promise(resolve => {
+          console.log('I AM A!');
+          setTimeout(() => {
+            aPreloadCalled = true;
+            console.log('I AM A after waiting!', aPreloadCalled);
+            resolve();
+          }, 10);
+        });
+      };
+
+      // b checks that a is setup first
+      const b = new ApplicationPlugin({ name: 'b', required: ['a'] });
+      b.preload = () => {
+        console.log('I AM B!', aPreloadCalled);
+        expect(aPreloadCalled).to.be.true;
+        return Promise.resolve();
+      }
+
+      Application.uses(b);
+      Application.uses(a);
+
+      const app = new Application();
+    });
```

Comment:
Good catch, fixed. At the time of writing, I was actually looking at the logs in the tests to check for order (which isn't really automated I guess 😅 ).

On that note, I think I need to revisit error handling during the preload chain since errors are being swallowed in an attempt to keep things flowing. Probably not a good idea.

Reply:
👍 

I agree we probably want errors from the preload chain to somehow notify the user and probably prevent the game from running. If a plugin the developer added failed to preload, they're probably just going to hit some weird bug or crash later anyway.